### PR TITLE
Moved try catch to cover more code

### PR DESCRIPTION
**Pull request by tristanbettany**
Client:	Geronimo Inns 1/GERONIMO
Job:	1/00037141 Geronimo - Ongoing Support
Phase:	004 January
Stage:	[95611] Instagram API Error

Try and catch more errors with larger try catch

### CODE REVIEW

File: tasks/instagrampoll.php [hunk 1]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
-		foreach($subscriptions as $sub) {
+			$subscriptions = \Propeller\Instagram\Model_Subscription::query()
+				->where('status', 'Live')
+				->get();
 
-			$tag = $instagram->getTag($sub->object_id);
-			$media = $tag->getMedia();
-			$count = 0;
+			foreach($subscriptions as $sub) 
+			{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 2]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
 
 	public function subscriptions()
 	{
-		$account = \Propeller\Instagram\Model_Account::query()
+		try 
+		{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 3]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
-
-				$image = \DB::select('instagram_id')
-					->from(\Propeller\Instagram\Model_Image::table())
-					->where('instagram_id', $med->id)
-					->execute()
-					->as_array('instagram_id', null);
-										
-				try
+				foreach($media as $med) 
 				{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 4]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
 					
-					if($content_type == 'application/xml' and $xml = new \SimpleXMLElement($output))
-					{
-						// Handle access denied error
-						if($xml->Code == 'AccessDenied')						
-						{						
-							throw new ImageAccessDeindedException("Access denied to 'http://distilleryimage0.s3.amazonaws.com/47040426c57211e2880f22000a1f9ca7_5.jpg'. This entry hasn't been saved to the DB.");
-						}
+						if($content_type == 'application/xml' and $xml = new \SimpleXMLElement($output))
+						{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 5]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
+							// Handle access denied error
+							if($xml->Code == 'AccessDenied')						
+							{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 6]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
-						$image->subscription_id = $sub->instagram_subscription_id;
-						$image->caption = $med->caption->text;
-
-						$image->save();
-
-						//Loop Thorugh Tags and store each one
-						foreach($med->tags as $tag) 
+						// Carry on..
+						if(!$image) 
 						{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 7]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
-							//Save Tags 
-							$image->tags[] = $tag_model;
 							$image->save();
-						}
 
-						$count++;
-					}				
+							//Loop Thorugh Tags and store each one
+							foreach($med->tags as $tag) 
+							{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 8]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
+								//Check Tag Existance
+								$check_tag_existance = \Propeller\Instagram\Model_Tag::query()->where('tag_name', $tag)->get_one();
+								if ($check_tag_existance) 
+								{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 9]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
+									//Get Model Of Tag if it already exists
+									$tag_id = $check_tag_existance->id;
+									$tag_model = \Propeller\Instagram\Model_Tag::find($tag_id);
+								}
+								else
+								{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 10]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
+								$image->save();
+							}
+
+							$count++;
+						}							
 				}
-				catch (ImageAccessDeindedException $e)
+
+				if($count) 
 				{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 11]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
+					$sub->last_image_received = time();
+					$sub->save();
+				}
+
+				\Cli::write($count ? $count.' updates for tag: '.$sub->object_id : 'No updates for tag: '.$sub->object_id, 'green');
+
+			}
+		}
+		catch (ImageAccessDeniedException $e)
+		{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 12]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
+			\Cli::write("Attempting to delete image", "cyan");
 					
-					if($image) 
-					{
-						$obj = \Propeller\Instagram\Model_Image::find($image->id);
+			if($image) 
+			{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 13]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
+				$obj = \Propeller\Instagram\Model_Image::find($image->id);
 						
-						// Delete the image as we no longer have access to it!
-						try
-						{
-							$obj->delete();				
+				// Delete the image as we no longer have access to it!
+				try
+				{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 14]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
-						}
-						catch(\Exception $e)
-						{
-							\Cli::error("Unable to delete image: " . $e->getMessage());
-						}
-					}				
+					\Cli::write("Image deleted", "green");
 				}
 				catch(\Exception $e)
 				{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: I know this bit didn't change, but could we keep all braces for anything other than functions and classes on the same line please?

File: tasks/instagrampoll.php [hunk 15]
```diff
@@ -60,134 +60,137 @@ public function cleanup()
...
-			}
-
-			\Cli::write($count ? $count.' updates for tag: '.$sub->object_id : 'No updates for tag: '.$sub->object_id, 'green');
-
+					\Cli::error("Unable to delete image: " . $e->getMessage());
+				}
+			}				
 		}
+		catch(\Exception $e)
+		{
```
Comment (xploSEoF) [inline tasks/instagrampoll.php]: Could we keep all braces for anything other than functions and classes on the same line please?

**Comment by tristanbettany** [comment]
No longer relevant

**Pull request closed without merging**